### PR TITLE
fix(sampler): avoid softmaxing the greedy one-hot into a near-uniform target

### PR DESCRIPTION
**Pull request by rebel-minkyu**
## Problem

`RBLNRejectionSamplerImpl.apply_sampling_constraints` returned a 0/1 one-hot
**probability** tensor for the all-greedy case, but its caller softmaxes the
result to build `target_probs`:

```python
logits = torch.zeros_like(logits).scatter_(-1, max_idx, 1.0)   # probabilities
...
target_probs = target_logits.softmax(dim=-1, dtype=torch.float32)   # caller
```

Softmaxing a 0/1 one-hot keeps the ordering but destroys the distribution: at
vocab 128k the greedy token lands at `p=2.1e-5` against `7.8e-6` for every other
token, i.e. 0.002% of the mass. At `p=2.1e-5` it rejected almost every draft, including
correct ones, and drew the recovered token near-uniformly from the whole
vocabulary.

## Fix

Return `-inf` everywhere and `0.0` at the argmax. Its softmax is an *exact*
one-hot, so a draft token is accepted only when it is the greedy choice, and any
recovered token is that greedy choice.

```python
return torch.full_like(logits, float("-inf")).scatter_(-1, max_idx, 0.0)
```

Only the `all_greedy` branch changes; random sampling is untouched.

## Verification

**Before**

```
"The capital of France is"                  -> ' Paris belirtilen นาง(di"\\ contamination � jose'
"The largest planet in the solar system is" -> ' the(em749_gb LLC depict_conditions<h'
"The chemical symbol for gold is"           -> ' AuTreeView METH.turn appendedapellido River.Dir'
"The first person to walk on the Moon was"  -> ' Neil FAR.setTo.Dir.GetName_supωμάτιο\theader'
```

**After**

```
"The capital of France is"                  -> ' Paris, and it is the most visited'
"The largest planet in the solar system is" -> ' the one that is closest to the sun'
"The chemical symbol for gold is"           -> ' Au. It is a soft, m'
"The first person to walk on the Moon was"  -> ' Neil Armstrong, who was the first man'
```

## How to test

```bash
pytest tests/torch_compile/e2e/v1/spec_decode/test_medusa_correctness.py -v
```

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
2. Verify output: `...`
3. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

**Comment by rebel-eunji** [comment]
Thanks for commit. However, we need to consider the cases when random requests and greedy requests are mixed.
I'll make extra patch for that.

**Comment by rebel-minkyu** [comment]
@rebel-eunji Thank you for the additional commit. As you mentioned, it would be worth checking the performance impact as a follow-up.

**Review by rebel-eunji — APPROVED** [review]
LGTM 👍
